internal/datasources/web.go

```diff
@@ -0,0 +1,53 @@
+package datasources
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+)
+
+type WebFileDatasource struct {
+	uri string
+}
+
+func NewWebFileDatasource(uri string) *WebFileDatasource {
+	return &WebFileDatasource{uri}
+}
+
+func (ds *WebFileDatasource) Load() (map[string]any, error) {
+	data := make(map[string]any)
+
+	res, err := http.Get(ds.uri)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	b, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	// b = []byte(`name = "toml"`)
+	// b = []byte(`{"name": "json"}`)
+	// b = []byte(`name: yaml`)
+	r := bytes.NewReader(b)
+
+	if IsJson(b) {
```

Comment:
Check the Content-Type header MIME type to know which library to use to unmarshal

Reply:
@orellazri 

I dug into it a bit

it seems like each type has a mime type:
```
application/json
application/yaml
application/toml
```

however, isn't that too limiting for the user? What if the toml/yaml source isn't sending back the exact mime-type we're handling, and instead of `application/yaml`, it sends back `text/plain` or anything that isn't the above, it would just fail?

actually I looked into it more, after writing the above paragraph and after researching more, and seems like in gomplate it's handled somewhat well.

it will assume the type based on the file extension -> or `Content-Type` header, and fail if isn't supported
however, if your source won't be interpreted correctly in the usual way, you'll be able to override the type assumption using a query param. in his case, something like `?type=application/json`

ref: https://docs.gomplate.ca/datasources/#overriding-mime-types

let me know what you think

Reply:
Not sure what I think about it. Customizing data sources with more parameters could be a good idea either way, but I think it's out of the scope of this PR. We can open a separate discussion after this is implementing, so that all datasources can be parameterized and not only external URL.
I think that for now this PR needs to detect the MIME type